slapo/utils/common.py

```diff
@@ -0,0 +1,55 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Common utilities used in schedule."""
+
+
+def get_hooks(mod):
+    """Get the hooks of a module.
+
+    Parameters
+    ----------
+    mod : torch.nn.Module
+        The module.
+
+    Returns
+    -------
+    dict
+        A dictionary of hooks.
+    """
+    hooks = {"fwd_pre": [], "fwd_post": [], "bwd_post": []}
+    for hook in mod._forward_hooks.values():
```

Comment:
the order of hooks is important. are these hooks a OrderedDict?

Reply:
Yes. Note that this is also the way PyTorch accesses these hooks, so if this causes any issues, then PyTorch would also have it.